perlock-core/src/main/java/name/mitterdorfer/perlock/SinglePathChangeListener.java

```diff
@@ -0,0 +1,18 @@
+package name.mitterdorfer.perlock;
+
+import java.nio.file.Path;
+import java.nio.file.WatchEvent;
+
+/**
+ * Callback interface for notifying the user when the chosen file changes.
+ * Suited for use as a functional interface
+ *
+ * <p>Implementation note: A <code>PathChangeListener</code> will be called from the internal <code>PathWatcher</code>
+ * thread. This implies that callback methods should return reasonably fast and offload heavy lifting to a dedicated
+ * thread. Otherwise, events may be lost.</p>
+ */
+public interface SinglePathChangeListener {
```

Comment:
Oh man, sorry for that. I have maybe time to look more closely into this but the most viable option seems to be a custom event class (this would also decouple the interface a bit from `WatchService` although I wouldn't worry too much about this here).

> One other thing that might be worth considering from an API point of view is OVERFLOW events.
> Currently perlock silently ignores them, but I think there should be some way for the user to be made aware of these.

Good suggestion. But I'd say this is a separate issue.